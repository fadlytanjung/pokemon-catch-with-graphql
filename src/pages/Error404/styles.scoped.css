.wrapper{
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100vh;
  width: 100vw;
  padding:48px 38px;
}
.wrapper img{
  max-width: 280px;
  width: 100%;
}
.wrapper h4{
  color: var(--text-color);
}
.wrapper h3{
  color: var(--black-color);
}
.wrapper button{
  margin-top: 12px;
}