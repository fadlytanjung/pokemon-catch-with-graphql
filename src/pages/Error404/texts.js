export default {
  ID: {
    BUTTON: 'Kembali ke beranda',
    TITLE: 'Halaman tidak ditemukan',
  },
  EN: {
    BUTTON: 'Back to home',
    TITLE: 'Page not found',
  },
};
