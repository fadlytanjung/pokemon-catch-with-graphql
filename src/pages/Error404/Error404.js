import React from 'react';
import { Link } from 'react-router-dom';
import Button from '../../components/elements/Button';
import Typography from '../../components/elements/Typography';
import MainBase from '../../components/layouts/MainBase';
import styles from './styles.scoped.css';
import texts from './texts';

export default function Error404() {
  const TEXTS = texts['EN'];
  return (
    <MainBase back>
      <section className={styles.wrapper}>
        <img src={'./assets/notfound.png'}/>
        <Typography tag="h3" variant="headline-medium" weight="semibold">{TEXTS.TITLE}</Typography>
        <Typography tag="h4" variant="button" weight="reguler">We cannot find your request page</Typography>
        <Link to="/">
          <Button variant="primary">
            {TEXTS.BUTTON}
          </Button>
        </Link>
      </section>
    </MainBase>
  );
}
