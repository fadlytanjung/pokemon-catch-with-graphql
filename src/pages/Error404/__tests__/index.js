import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Error404 from '../Error404';

describe('src/pages/Error404', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Error404 />);
    expect(tree).toMatchSnapshot();
  });
});
