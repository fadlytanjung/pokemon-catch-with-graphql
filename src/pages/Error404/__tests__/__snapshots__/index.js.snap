// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/Error404 render 1`] = `
<MainBase
  back={true}
  label=""
  to="/"
>
  <section>
    <img
      src="./assets/notfound.png"
    />
    <Typography
      tag="h3"
      variant="headline-medium"
      weight="semibold"
    >
      Page not found
    </Typography>
    <Typography
      tag="h4"
      variant="button"
      weight="reguler"
    >
      We cannot find your request page
    </Typography>
    <Link
      to="/"
    >
      <Button
        className=""
        disabled={false}
        height={48}
        onClick={null}
        to=""
        type="button"
        variant="primary"
        width={168}
      >
        Back to home
      </Button>
    </Link>
  </section>
</MainBase>
`;
